fontes/avaliador-sintatico/avaliador-sintatico.ts

```diff
@@ -415,8 +416,27 @@ export class AvaliadorSintatico implements AvaliadorSintaticoInterface {
         /*if (!(valorModificador instanceof Metodo)) {
             quantificador = this.avancarEDevolverAnterior();
         }*/
-        if (valorModificador.hasOwnProperty('tipo') && valorModificador.tipo === tiposDeSimbolos.NUMERO) {
-            quantificador = this.avancarEDevolverAnterior();
+        if (valorModificador.hasOwnProperty('tipo') && valorModificador.tipo === tiposDeSimbolos.NUMERO) {       
+
+            // IF - Se for um valor fora da lista de valores numéricos, executa o trecho.
+            // ELSE - Se estiver na lista de valores numéricos, não executa esse trecho.
+            if(!(ValorNumerico.includes(modificador.lexema))) {
+                // IF - Se estiver na segunda lista, verifica a existência do quantificador
+                // ELSE - Se não estiver, busca direto o quantificador
+                if (ValorNumericoComQuantificador.includes(modificador.lexema)) {
+                    const encontrarQuantificador = this.simbolos.find((simbolo) => 
```

Comment:
Acho perigoso fazer assim, sobretudo se for um arquivo enorme. O certo seria apenas verificar o próximo símbolo. Se for um quantificador, guarda na variável.